Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertMaterialQuantities.cs

```diff
@@ -65,6 +99,78 @@ public IEnumerable<Objects.Other.MaterialQuantity> MaterialQuantitiesToSpeckle(D
     }
 
     #endregion
+    /// <summary>
+    /// Revit API default methods for computing volumes and areas is malfunctioning for some objects
+    /// 
+    /// </summary>
+    /// <param name="element"></param>
+    /// <returns>true f the element's quantities needs to be computetd based on the geometry.returns>
+    private bool RequiresGeometryComputation(DB.Element element)
+    {
+      if (element.Category.Id.IntegerValue == (int)BuiltInCategory.OST_Windows || 
+        element.Category.Id.IntegerValue == (int)BuiltInCategory.OST_Doors ||
+        element is DB.Architecture.Railing||
+        element is DB.Architecture.TopRail) return true;
+      return false;
+    }
```

Comment:
Could you explain a bit why you are using these specific clauses to determine if an object needs geometry computation? Were these just the elements you ran into that needed it? If so, how did you know these elements needed it

Reply:
Hi @connorivy

thanks for the feedback. I am using this specific categories to start with. I am quite sure that there are others but in this way they can be easily added. I introduced the issue previously that the Revit API handles the computation of the areas differently. While for walls, the default computation returns the area of the largest surface (or the area along the LocationCurve multiplied with the height), for other categories, the default method returns the sum of all surfaces. 

For now, I chose Windows, doors railing and top rail as those are the ones where we noticed the inconsistency. I am working on a robust testing method where I am comparing the computed quantities to the expected quantities. This should give an overview. 

I know that these categories are inconsisten as we are testing the quantities for our use cases.